src/cli/cli_link_metrics.cpp

```diff
@@ -206,15 +206,22 @@ template <> otError LinkMetrics::Process<Cmd("mgmt")>(Arg aArgs[])
             }
         }
 
+        index = 4;
```

Comment:
I think parsing with pre-set index (we assume which position is what argument) makes it difficult to read and maintain the code. Once we have a change, it's error-prone and not easy to update.

My suggestion is that we use 'sync {0/1}' instead of appending 'sync' at the end. In this way, we can do the parsing more stateless. I suggest adding it right after `mgmt <ipaddr>`. If no 'sync {0/1}' then it's regarded as non-blocking. Otherwise it's set to the value. And the upcoming arguments are deferred correspondingly.

As an example, 
```
linkmetrics query fe80:0:0:0:3092:f334:1455:1ad2 sync 1 forward 1 dra pqmr

or

linkmetrics query fe80:0:0:0:3092:f334:1455:1ad2 sync 0 forward 1 dra pqmr

or

linkmetrics query fe80:0:0:0:3092:f334:1455:1ad2 forward 1 dra pqmr  // Default, non-blocking
```

I admit the existing parsing code isn't well written. Sorry for the inconvenience.

Reply:
Update:

Actually the key point here isn't using `sync {0/1}` or `sync`, but the position of `sync`. So no need to parse one more args. We can just to use `sync`.

Reply:
Updated